diff --git a/src/Stack/Types/GhcPkgId.hs b/src/Stack/Types/GhcPkgId.hs
index 0d06b379..9dc9e5d8 100644
--- a/src/Stack/Types/GhcPkgId.hs
+++ b/src/Stack/Types/GhcPkgId.hs
@@ -60,7 +60,7 @@ parseGhcPkgId x = go x
 -- | A parser for a package-version-hash pair.
 ghcPkgIdParser :: Parser GhcPkgId
 ghcPkgIdParser =
-    let elements =  "_.-" :: String in
+    let elements =  "_.-+" :: String in
     GhcPkgId . T.pack <$> many1 (choice [digit, letter, satisfy (`elem` elements)])
 
 -- | Get a string representation of GHC package id.
